### ISSUE TITLE
Angular 1.4.9 Intro's Regression in ngOfficeUiFabric

### ISSUE THREAD

**Issue by andrewconnell** [ngOfficeUIFabric/ng-officeuifabric#139]
#### TL;DR

It looks like we’ve gotten away with some stuff so far, until Angular release 1.4.9 and added a check that introduced a regression for us. @s-kainet found it and bugged it (https://github.com/angular/angular.js/issues/13832) but apparently not a bug. It looked like a bug because they had a gap in their compiler that didn't check for this... so we just go away with it.  We need to fix this to support any future versions of Angular without errors.

The issue is that Angular was not checking for the presence of attributes on directives when the directive defintiion specified the property was manditory with a `=`. If they are optional parameters we should use `=?`.
#### Repo Steps

Today our test suite passes unless you update to Angular 1.4.9 in which case 5 tests fail in the uif-contextual-menu directive. If you run the demos you see many more directives have issues.
#### Actions To Take

For the directives without failing tests but issues throwing errors should get tests that reflect this failure, then the directive definition should be updated to pass this test.

Ideally this could be addressed at the same time as the other regression we introduced removing jQuery from our test suites #121.
#### Directives affected
- [x] `uif-callout` @jjczopek
- [x] `uif-contextual-menu` @s-KaiNet
- [x] `uif-searchbox` @jigargandhi
- [x] `uif-toggle` @rolandoldengarm

**Comment by jjczopek** [comment]
@andrewconnell : from what I can tell, `uif-callout` didn't have failing tests due to this regression and I found no errors in the demo (I double checked if I use 1.4.9). Do I still need to introduce test for this? If yes, how do I do that? Or is it OK if I just mark property on scope as `=?` ?
